sql.wit.md

```diff
@@ -0,0 +1,66 @@
+# `wasi-sql` API
+
+## Interfaces
+
+```wit
+interface "wasi:sql" {
+    // iterator item type
+    resource row {
+        field-name: list<string>
+        values: list<data-types>
+        index: u32
+    }
+    
+    // common data types
+    variant data-types {
+        integer(s32),
+        float(float64),
+        str(string),
+        boolean(bool),
+        date(string),
```

Comment:
That's a good point. Initially, I was thinking of just relying on the format used by the underlying database. On the other hand, it might be a good idea to define a standardized format for storing dates in the interface itself for consistency's sake — we could use ISO 8601, or smt of the sort. That said, I'm not sure. How do you feel about just relying on the underlying format?

Reply:
I'm not sure what "the underlying format" is.  When I ask SQL Server for a `datetimeoffset` column, I assume it is stored in some opaque binary format, and surfaced as a `System.DateTimeOffset` or `chrono::DateTime` or whatever type by the driver.  So we would probably need to format it anyway.

I could be wrong though - could you expand on your understanding of "underlying string format"?

Reply:
@itowlson ~ I tried out implementing this interface for postgres, and here's what I did w/ regard to this:

```rs
                        "date" => {
                            let v: String = row.get(i);
                            let parsed = NaiveDate::parse_from_str(&v, "%Y-%m-%d").unwrap();
                            DataType::Date(parsed.to_string())
                        }
                        "time" => {
                            let v: String = row.get(i);
                            let parsed = NaiveTime::parse_from_str(&v, "%H:%M:%S").unwrap();
                            DataType::Time(parsed.to_string())
                        }
                        "timestamp" => {
                            let v: String = row.get(i);
                            let parsed =
                                NaiveDateTime::parse_from_str(&v, "%Y-%m-%d %H:%M:%S").unwrap();
                            DataType::Timestamp(parsed.to_string())
                        }
```